### PR TITLE
Servlet for admins to upload files to data/settings/ 

### PR DESCRIPTION
**Pull request by DravitLochan**
second part of  #112 

Changes: Added servlet for upload of file
test locally at -> 127.0.0.1:4000/data/settings

### CODE REVIEW

**Comment by DravitLochan** [comment]
@jyoth1raditya @dynamitechetan @chashmeetsingh Please review!

**Comment by DravitLochan** [comment]
OK. Updating it..

**Comment by DravitLochan** [comment]
Sorry. I was out due to family reasons. Will do it asap.

**Comment by DravitLochan** [comment]
@Orbiter Updated the pr

**Comment by DravitLochan** [comment]
@mariobehling Please merge!

File: src/ai/susi/server/api/aaa/UploadDataSettings.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import java.io.IOException;
+
+/**
+ * Created by dravit on 12/6/17.
+ * Servlet to upload file to data/settings folder
+ */
+public class UploadDataSettings extends AbstractAPIHandler implements APIHandler{
+    @Override
+    public String getAPIPath() {
+        return "/data/settings/upload";
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/UploadDataSettings.java]: We should not have "phantasy api paths". All servlets have paths according to their class package. In this case the API path must be "/aaa/uploadDataSettings.json".

File: src/ai/susi/server/api/aaa/UploadDataSettings.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        String fileName = (String) post.getRequest().getParameter("file");
+        if (fileName == null || fileName.equals(""))
+            try {
+                throw new ServletException("Invalid or non-existent file parameter in SendWord servlet.");
+            } catch (ServletException e) {
+                e.printStackTrace();
+            }
+        // add the .doc suffix if it doesn't already exist
+        if (fileName.indexOf(".json") == -1)
+            fileName = fileName + ".json";
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/UploadDataSettings.java]: this does not make sense for non-json files. Remove this check and replace it with a test if the file with the given name exists.

File: src/ai/susi/server/api/aaa/UploadDataSettings.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+        String wordDir = getServletContext().getInitParameter("word-dir");
+        if (wordDir == null || wordDir.equals(""))
+            try {
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/UploadDataSettings.java]: What is this used for?

File: src/ai/susi/server/api/aaa/UploadDataSettings.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                result.put("success","false");
+                e.printStackTrace();
+            }
+        ServletOutputStream stream = null;
+        BufferedInputStream buf = null;
+        try {
+            stream = response.getOutputStream();
+            File file = new File(filePath +  fileName);
+            response.setContentType("application/json");
+            response.addHeader("Content-Disposition", "attachment; filename="
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/UploadDataSettings.java]: Why do you modify the response object? The servlet always returns a ServletResponse object which handles response types. Remove this.

File: src/ai/susi/server/api/aaa/UploadDataSettings.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    + fileName);
+            response.setContentLength((int) file.length());
+            FileInputStream input = new FileInputStream(file);
+            buf = new BufferedInputStream(input);
+            int readBytes = 0;
+            while ((readBytes = buf.read()) != -1)
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/UploadDataSettings.java]: all you do here is to write into the response stream. But this servlet is about the upload of data into settings, right? This is implementing a download, not a upload.

Comment (DravitLochan) [inline src/ai/susi/server/api/aaa/UploadDataSettings.java]: Modifying the servlet. Sorry for the inconvenience caused!

**Comment by DravitLochan** [comment]
this is a little messed up. Closing this pr. Will open a new one!

**Pull request closed without merging**